### PR TITLE
feat(inputs): commit one InputBasis and snapshot fence as input

### LINKED ISSUE #188: Commit one coherent input basis per request
## Problem

Input loading and snapshots are fragmented across consumers, so one request can observe files from different revisions or perform untracked filesystem reads. This causes torn semantic results, stale diagnostics, and nondeterministic incremental behavior.

## Expected outcome

Each request operates on one committed `InputBasis` guarded by a snapshot fence, and every consumer observes the same source revisions and negative facts.

## Acceptance

- A request binds exactly one immutable `InputBasisId` before semantic work begins.
- Every source read is represented in the committed input basis rather than performed by a consumer.
- `SnapshotFence` prevents publication from mixing revisions.
- Missing inputs are recorded as explicit negative facts.
- Repeated execution over the same basis produces deterministic results and provenance.

AI-Generated

### PR DESCRIPTION
**Pull request by pikax**
Readiness comes only from trusted implementation-ledger rows. A READY node may start; tooling does not validate commit locators, Git identity, receipts, leases, external state, or runtime admission.

One committed input and coherent snapshot authority. The current owner is **fragmented input loading and snapshots**. The final and sole owner is **one committed InputBasis with deterministic load waves**. This charter accepts one authority/migration/cutover boundary; it contains no independently dispatchable subblocks.

### Scope

- Production surfaces: `crates/verter_analysis_inputs/src`, `crates/verter_session/src`.
- Named API/data boundaries: `InputBasis`, `LoadWave`, `NegativeFact`, `SnapshotFence`.
- Mutation boundary: only the production surfaces and named API/data boundaries above; every changed path must be inside both that charter surface and the acquired conflict domain, and sibling ownership is excluded.

### Acceptance

Preflight evidence selection: preserve all four acceptance outcomes below, then select the smallest evidence set that actually discriminates the touched contract. Existing behavioral coverage, compiler/type/capability enforcement, static validation, canonical gates, bounded inspection, and benchmarks are valid when accompanied by a terse rationale.

- **F1-AC1 — sole-owner outcome:** the named final owner must be sole and every displaced route named below must be deleted or structurally rejected. Prefer existing type, capability, dependency, compiler, or static enforcement. Add a negative or mutation test only for a plausible critical fail-closed/correctness boundary or a reproduced defect that existing evidence does not discriminate.
- **F1-AC2 — positive contract:** the named API/data boundary must preserve exact identities, provenance, completeness, and deterministic ordering. Reuse existing coverage or extend/table-drive one test before creating a new test.
- **F1-AC3 — incremental equivalence:** when the changed scope owns or affects incremental, cache, cancellation, stale-publication, or partial-result authority, prove incremental equals fresh and degraded outcomes cannot warm. Otherwise record a terse not-applicable rationale tied to the untouched authority.
- **F1-AC4 — bounded work:** when the changed scope owns or affects a hot path, prove no hidden duplicate parse, resolve, plan, emit, copy, allocation, or retained candi…

Closes #188

## Summary by CodeRabbit

- **New Features**
  - Added committed input snapshots for files, directories, and explicitly absent observations.
  - Added content-based snapshot identities and validation for inconsistent observations.
  - Added snapshot fences to prevent publication from mismatched snapshots.
  - Added session request binding for a single committed snapshot.
  - Route analysis now uses committed snapshots while preserving absent-file information.

- **Tests**
  - Added coverage for snapshot identity, immutability, conflicts, unrecorded reads, request binding, and snapshot consistency.

### CODE REVIEW

**Comment by pikax** [comment]
@coderabbitai review

**Comment by pikax** [comment]
@coderabbitai review

File: crates/verter_analysis_inputs/src/input_basis.rs [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+            kind: ObservationKind::File { content },
+        }
+    }
+
+    /// Directory observation. Revision is the digest of the sorted listing.
+    #[must_use]
+    pub fn directory(canonical: impl Into<Arc<str>>, mut entries: Vec<DirectoryEntry>) -> Self {
+        entries.sort();
+        entries.dedup();
+        let revision = directory_revision(&entries);
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_analysis_inputs/src/input_basis.rs]: Recorded on the TAMA debt ledger as `debt_coderabbit_node_4151fc92af15f259fea08e3c84521c37_coderabbit-4045931284_mu6xordr` (P2); not repaired in this PR. The thread is resolved so the pull request does not wait on it; the debt row is inherited by the next node that touches these files.

File: crates/verter_analysis_inputs/src/input_basis.rs [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+                return Err(CommitError::OverlappingPositiveAndNegative { canonical });
+            }
+            negative_map.insert(canonical, fact);
+        }
+
+        let id = InputBasisId::from_canonical(&InputBasisDescriptor {
+            wave: &wave,
+            observations: &observation_map,
+            negatives: &negative_map,
+        });
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_analysis_inputs/src/input_basis.rs:292]: Recorded on the TAMA debt ledger as `debt_coderabbit_node_4151fc92af15f259fea08e3c84521c37_coderabbit-4045931302_mu6xordr` (P2); not repaired in this PR. The thread is resolved so the pull request does not wait on it; the debt row is inherited by the next node that touches these files.

File: crates/verter_session/src/request_context.rs [hunk 3]
```diff
@@ -1157,6 +1157,9 @@ pub struct RequestContext {
     /// request, shared across the worker threads the request fans out
     /// to via `install_tls`).
     pub dep_canonical_memo: parking_lot::Mutex<rustc_hash::FxHashMap<String, String>>,
+    /// Sole committed input for this request. Bound once by
+    /// [`Self::bind_committed_input`] before semantic work.
+    committed_input: std::sync::OnceLock<crate::input_basis::RequestInputBinding>,
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_session/src/request_context.rs:1162]: Recorded on the TAMA debt ledger as `debt_coderabbit_node_4151fc92af15f259fea08e3c84521c37_coderabbit-4045931308_mu6xordr` (P2); not repaired in this PR. The thread is resolved so the pull request does not wait on it; the debt row is inherited by the next node that touches these files.

File: crates/verter_session/src/route_analysis_inputs.rs [hunk 4]
```diff
@@ -24,101 +24,146 @@
...
-    if let Some(content) = workspace.read_file(&pkg_path) {
-        inputs.insert_file(pkg_path, content);
-    }
+    capture.probe_file(workspace, pkg_path);
 
-    // Framework detection itself is pure — run it against the snapshot
-    // we've captured so far, exactly mirroring what `build_route_analysis`
-    // will do with the SAME inputs later.
-    let framework = detect_routing_framework(&inputs, project_root);
+    let framework = detect_routing_framework(&capture.detection_inputs(), project_root);
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_session/src/route_analysis_inputs.rs:94]: Recorded on the TAMA debt ledger as `debt_coderabbit_node_4151fc92af15f259fea08e3c84521c37_coderabbit-4045931319_mu6xordr` (P2); not repaired in this PR. The thread is resolved so the pull request does not wait on it; the debt row is inherited by the next node that touches these files.

File: crates/verter_session/src/route_analysis_inputs.rs [hunk 5]
```diff
@@ -24,101 +24,146 @@
...
     match framework {
         RoutingFramework::NuxtPages => {
-            walk_dir_recursive(workspace, &format!("{trimmed_root}/pages"), &mut inputs);
+            capture.walk_dir(workspace, &format!("{trimmed_root}/pages"));
         }
         RoutingFramework::UnpluginVueRouter => {
             let src_pages = format!("{trimmed_root}/src/pages");
             if workspace.is_dir(&src_pages) {
-                walk_dir_recursive(workspace, &src_pages, &mut inputs);
+                capture.walk_dir(workspace, &src_pages);
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_session/src/route_analysis_inputs.rs]: Recorded on the TAMA debt ledger as `debt_coderabbit_node_4151fc92af15f259fea08e3c84521c37_coderabbit-4045931330_mu6xordr` (P2); not repaired in this PR. The thread is resolved so the pull request does not wait on it; the debt row is inherited by the next node that touches these files.

File: crates/verter_session/src/route_analysis_inputs.rs [hunk 6]
```diff
@@ -24,100 +24,203 @@
...
+                ctx.committed_input()
+                    .expect("request bind stored the route-analysis basis"),
+            ),
+            Err(rejected) => {
+                let bound = ctx
+                    .committed_input()
+                    .expect("failed bind implies a stored request basis");
+                bound
+                    .admit_publication(rejected.basis())
+                    .expect("route-analysis basis must match the request snapshot fence");
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_session/src/route_analysis_inputs.rs:65]: Recorded on the TAMA debt ledger as `debt_coderabbit_node_4151fc92af15f259fea08e3c84521c37_coderabbit-4046795085_mu6y6556` (P2); not repaired in this PR. The thread is resolved so the pull request does not wait on it; the debt row is inherited by the next node that touches these files.

File: crates/verter_session/src/route_analysis_inputs.rs [hunk 7]
```diff
@@ -24,100 +24,203 @@
...
+    /// Walk `dir`. Returns whether a directory listing was recorded.
+    /// Confirmed non-directories become [`NegativeFact::absent`]. A `read_dir`
+    /// error aborts capture rather than being classified as absence.
+    fn walk_dir(&mut self, workspace: &dyn verter_workspace::WorkspaceRead, dir: &str) -> bool {
+        if self.directory_read_failed {
+            return false;
+        }
+        if !workspace.is_dir(dir) {
+            self.keys.push(dir.to_string());
+            self.negatives.push(NegativeFact::absent(dir));
```
[the remark this answers is not in the document]
Comment (pikax) [reply crates/verter_session/src/route_analysis_inputs.rs:181]: Recorded on the TAMA debt ledger as `debt_coderabbit_node_4151fc92af15f259fea08e3c84521c37_coderabbit-4046795097_mu6y6556` (P2); not repaired in this PR. The thread is resolved so the pull request does not wait on it; the debt row is inherited by the next node that touches these files.
